elastica/contact_forces.py

```diff
@@ -423,6 +424,143 @@ def _calculate_contact_forces_self_rod(
                 external_forces_rod[..., j + 1] += net_contact_force
 
 
+def _calculate_contact_forces_rod_sphere(
+    x_collection_rod,
+    edge_collection_rod,
+    x_sphere_center,
+    x_sphere_tip,
+    edge_sphere,
+    radii_sum,
+    length_sum,
+    internal_forces_rod,
+    external_forces_rod,
+    external_forces_sphere,
+    external_torques_sphere,
+    sphere_director_collection,
+    velocity_rod,
+    velocity_sphere,
+    contact_k,
+    contact_nu,
+    velocity_damping_coefficient,
+    friction_coefficient,
+) -> None:
+    # We already pass in only the first n_elem x
+    n_points = x_collection_rod.shape[1]
+    sphere_total_contact_forces = np.zeros((3))
+    sphere_total_contact_torques = np.zeros((3))
+    for i in range(n_points):
+        # Element-wise bounding box
+        x_selected = x_collection_rod[..., i]
+        # x_sphere is already a (,) array from outside
+        del_x = x_selected - x_sphere_tip
+        norm_del_x = _norm(del_x)
+
+        # If outside then don't process
+        if norm_del_x >= (radii_sum[i] + length_sum[i]):
+            continue
+
+        # find the shortest line segment between the two centerline
+        distance_vector, x_sphere_contact_point, _ = _find_min_dist(
+            x_selected, edge_collection_rod[..., i], x_sphere_tip, edge_sphere
+        )
+        distance_vector_length = _norm(distance_vector)
+        distance_vector /= distance_vector_length
+
+        gamma = radii_sum[i] - distance_vector_length
+
+        # If distance is large, don't worry about it
+        if gamma < -1e-5:
+            continue
+
+        rod_elemental_forces = 0.5 * (
+            external_forces_rod[..., i]
+            + external_forces_rod[..., i + 1]
+            + internal_forces_rod[..., i]
+            + internal_forces_rod[..., i + 1]
+        )
+        equilibrium_forces = -rod_elemental_forces + external_forces_sphere[..., 0]
+
+        normal_force = _dot_product(equilibrium_forces, distance_vector)
+        # Following line same as np.where(normal_force < 0.0, -normal_force, 0.0)
+        normal_force = abs(min(normal_force, 0.0))
```

Comment:
Normal force is not used, lets remove these lines. Can you also check other classes as well?

Reply:
removed from rod-sphere and rod-cylinder, need to remove from rod-rod contact and to retune the rod-rod examples accordingly